### DISCUSSION TITLE
I always get stale user data although my Users table is updated

### DISCUSSION THREAD

**Discussion by dgknca** [supabase/supabase#12965] [Questions] [answered]
Although my Users table has been updated (PUT request returns 200 and I see the field is updated on Supabase dashboard), I still get stale user data in my NextJS application. How can I make it pull the updated data? Even if I refresh the page, or even stop and restart my development server, I still get the old data. What should I do?

```js
import { createBrowserSupabaseClient } from '@supabase/auth-helpers-nextjs';
import { type Session, SessionContextProvider } from '@supabase/auth-helpers-react';
// ...

function MyApp({
  Component,
  pageProps,
}: AppProps<{
  initialSession: Session;
}>) {
  const [supabaseClient] = React.useState(() =>
    createBrowserSupabaseClient({
      supabaseUrl: process.env.NEXT_PUBLIC_SUPABASE_URL,
      supabaseKey: process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY,
    })
  );

  return (
      <SessionContextProvider
        supabaseClient={supabaseClient}
        initialSession={pageProps.initialSession}
      >
        <Component {...pageProps} />
      </SessionContextProvider>
  );
}

export default MyApp;
```

(I don't use tanstack query or SWR. So there shouldn't be cache.)

**Comment by GaryAustin1** [comment, marked as answer]
Doing a refreshSession() call should pull down the latest jwt.

**Reply by dgknca** [reply]
I've spent hours on this since yesterday. you saved my day. thank you so much.
